torch/csrc/autograd/variable.cpp

```diff
@@ -553,15 +573,16 @@ void VariableHooks::retain_grad(const at::TensorBase& self) const {
         var->mutable_grad() = var->grad() + grad;
       }
     }
+    return at::TensorBase{};
```

Comment:
What is the contract on the returned value here?

Reply:
If the return value is defined, then it will modify the gradient; otherwise if it is not-defined, then this hook is considered as only "reading" the value of the gradient.

See: https://github.com/pytorch/pytorch/blob/645fb217c06348a4f1ccdf68a93bd711f7158c62/torch/csrc/autograd/cpp_hook.cpp#L38-L40